pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -93,124 +117,172 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	}
 	labels["app"] = el.Name
 
-	// Create the EventListener Deployment
-	deployment := &appsv1.Deployment{
-		ObjectMeta: metav1.ObjectMeta{
-			// Create the EventListener's Deployment in the same Namespace as where
-			// the EventListener was created
-			Namespace: el.Namespace,
-			// Give the Deployment the same name as the EventListener
-			Name: el.Name,
-			// If our EventListener is deleted, then its Deployment should be as well
-			OwnerReferences: []metav1.OwnerReference{*el.GetOwnerReference()},
-			Labels:          labels,
-		},
-		Spec: appsv1.DeploymentSpec{
-			Selector: &metav1.LabelSelector{
+	oldService, err := c.KubeClientSet.CoreV1().Services(el.Namespace).Get(el.Name, metav1.GetOptions{})
+	switch {
+	case err == nil:
+		el.Status.SetExistsCondition(v1alpha1.ServiceExists, nil)
+		el.Status.Configuration.Hostname = listenerHostname(el.Name, el.Namespace)
```

Comment:
Within `Reconcile()`, I set the deep copy to an empty status and rebuild it to compare against. I was thinking about certain edge cases like if you were delete a condition on the `EventListener`, it wouldn't be able to get reconciled if it wasn't idempotent like this. I'll take a look and see if it can be simplified.